python/ray/data/_internal/datasource/parquet_datasource.py

```diff
@@ -278,8 +279,10 @@ def __init__(
         self._to_batches_kwargs = to_batch_kwargs
         self._columns = columns
         self._schema = schema
-        self._file_metadata_shuffler = None
         self._include_paths = include_paths
+
+        _validate_shuffle_arg(shuffle)
```

Comment:
I wasn't sure what the best way to organize this was, I expected `ParquetDatasource` to extend `FileBasedDatasource`.

Reply:
They have different code paths because `ParquetDatasource` leverages PyArrow's ParquetDataset